contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -602,18 +617,20 @@ grant_revoke_dbo_to_login(const char* login, const char* db_name, bool is_grant)
 	
 	initStringInfo(&query);
 
-	dbo = lappend(dbo, make_accesspriv_node(dbo_role_name));
+	rolelist = lappend(rolelist, make_accesspriv_node(role_name));
 
 	if (is_grant)
 	{
 		/* Build dummy GRANT statement to grant membership to login  */
-		appendStringInfo(&query, "GRANT dummy TO dummy; ");
+		appendStringInfo(&query, "GRANT dummy TO dummy ");
 	}
 	else
 	{
 		/* Build dummy REVOKE statement to revoke membership from login */
-		appendStringInfo(&query, "REVOKE dummy FROM dummy; ");
+		appendStringInfo(&query, "REVOKE dummy FROM dummy ");
 	}
+	if (grantor)
+		appendStringInfo(&query, "GRANTED BY %s; ", grantor);
```

Comment:
Let's follow the existing way to fill the placeholder for grantor after parsing to avoid SQL injection. `GRANTED BY  dummy` instead of `GRANTED BY %s;`.